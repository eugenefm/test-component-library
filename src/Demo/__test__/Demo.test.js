import React from 'react';
import ReactDom from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

import Demo from '../Demo';

afterEach(cleanup);

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDom.render(<Demo />, div);
	ReactDom.unmountComponentAtNode(div);
});

it('render Demo Component correctly', () => {
	const { getByTestId } = render(<Demo type={'success'}>Hello World</Demo>);
	expect(getByTestId('DemoMessage')).toHaveTextContent('Hello World');
});

it('matches snapshot', () => {
	const tree = renderer.create(<Demo type={'danger'}>Errors</Demo>).toJSON();
	expect(tree).toMatchSnapshot();
});
