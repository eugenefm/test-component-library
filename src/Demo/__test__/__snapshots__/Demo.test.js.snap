// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdVaJa ePuhSX"
  data-testid="DemoMessage"
  type="danger"
>
  Errors
</div>
`;
